src/rpc.rs

```diff
@@ -126,24 +144,42 @@ impl RpcSol for RpcSolImpl {
     fn get_transaction_count(&self, meta: Self::Metadata) -> Result<u64> {
         meta.request_processor.get_transaction_count()
     }
-    // fn send_transaction(&self, meta: Self::Metadata, to: String, tokens: i64) -> Result<String> {
-    //     let client_keypair = read_keypair(&meta.keypair_location.unwrap()).unwrap();
-    //     let mut client = mk_client(&meta.leader.unwrap());
-    //     let last_id = client.get_last_id();
-    //     let to_pubkey_vec = bs58::decode(to)
-    //         .into_vec()
-    //         .expect("base58-encoded public key");
-    //
-    //     if to_pubkey_vec.len() != mem::size_of::<Pubkey>() {
-    //         Err(Error::invalid_request())
-    //     } else {
-    //         let to_pubkey = Pubkey::new(&to_pubkey_vec);
-    //         let signature = client
-    //             .transfer(tokens, &client_keypair, to_pubkey, &last_id)
-    //             .unwrap();
-    //         Ok(bs58::encode(signature).into_string())
-    //     }
-    // }
+    fn request_airdrop(&self, meta: Self::Metadata, id: String, tokens: u64) -> Result<bool> {
+        let pubkey_vec = bs58::decode(id)
+            .into_vec()
+            .map_err(|_| Error::invalid_request())?;
+        if pubkey_vec.len() != mem::size_of::<Pubkey>() {
+            return Err(Error::invalid_request());
+        }
+        let pubkey = Pubkey::new(&pubkey_vec);
+        let previous_balance = meta
+            .request_processor
+            .get_balance(pubkey)
+            .map_err(|_| Error::internal_error())?;
+        request_airdrop(&meta.drone_addr, &pubkey, tokens).map_err(|_| Error::internal_error())?;
+        let now = Instant::now();
+        let mut balance;
+        loop {
+            balance = meta
+                .request_processor
+                .get_balance(pubkey)
```

Comment:
This seems error prone. Can you poll for the signature instead?

Reply:
How about we just make the caller of the RPC API poll their balance?   But yeah, if we could get the tx signature back from the drone then passing it back here would be super

Reply:
Yeah, I was pretty ambivalent about putting this in the request_airdrop handler vs. polling on the client side.
Returning the tx signature from the drone is a better solution, but maybe a separate PR?

Reply:
Is something important blocked by this PR?

Reply:
/me is not blocked

Reply:
Okay, no problem. Let's get this one merged.